File: MyEngine/TestingCrossPlatformGame/TestingCrossPlatformGame.Core/Content/Scripts/PlayerScript.cs
using Microsoft.Xna.Framework;
using MyEngine.MyCore.MyEntities;
using MyEngine.MyCore.MyComponents;
using MyEngine.Scripting;
using System;

namespace TestingCrossPlatformGame.Core.Content.Scripts
{
    public class PlayerScript : IScript
    {
        public int EntityId { get; set; }

        public void Initialize(MainEntity entity)
        {
            EntityId = entity.Id;
            entity.AddComponent(new TransformComponent(Vector2.Zero, 4, Vector2.One)); 
        }

        public void Update(GameTime gametime, MainEntity entity)
        {
            // Ejemplo: si la entidad tiene un componente "Health"
            if (!entity.HasComponent<TransformComponent>())
            {
                TransformComponent transform = entity.GetComponent<TransformComponent>();

                

            }
            else
            {
                
            }
        }

        public void Cleanup()
        {
            GC.Collect();
        }

    }


}

File: MyEngine/F#Scripts/Scripts/PlayerScript.fs
namespace TestingCrossPlatformGame.Core.Content.Scripts

open Microsoft.Xna.Framework
open MyEngine.MyCore.MyEntities
open MyEngine.MyCore.MyComponents
open MyEngine.Scripting
open System


type PlayerScript() =
    interface IScript with
        member this.EntityId
            with get (): int = 
                raise (System.NotImplementedException())
            and set (v: int): unit = 
                raise (System.NotImplementedException())
   


        member this.Initialize(entity: MainEntity) =
            entity.AddComponent(new TransformComponent(Vector2.Zero, 4f, Vector2.One))


        member this.Update(gameTime: GameTime, entity: MainEntity) =
            if entity.HasComponent<TransformComponent>() then
                let transform = entity.GetComponent<TransformComponent>()
                // Aquí pondrías lógica para mover/actualizar transform
                ()
            else
                ()

        member this.Cleanup() =
            GC.Collect()



